overlord/ifacestate/handlers.go

```diff
@@ -501,6 +501,12 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 		logger.Debugf("Connect handler: skipping setupSnapSecurity for snaps %q and %q", plug.Snap.InstanceName(), slot.Snap.InstanceName())
 	}
 
+	// for undo handler. it's only relevant for preserving undesired flag,
+	// otherwise undo can delete any trace of the connection.
+	if old, ok := conns[connRef.ID()]; ok && old.Undesired {
+		task.Set("old-conn", old)
```

Comment:
If the goal is to remember the undesired flag and nothing else then perhaps we should create an empty connection state with just the Undesired field populated from the old state.

Reply:
We should restore previous conn state for completness because of a few attributes of conns (bygadget, autoconnect, hotplug key) that should be kept; what I meant by this comment was that we only care about preserving conn if undesired is set, as otherwise we should simply drop the record from conns. I'll reword the comment.